src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/Razor/GenerateMethodCodeActionResolver.cs

```diff
@@ -138,19 +161,41 @@ private WorkspaceEdit GenerateMethodInCodeBlock(RazorCodeDocument code, Generate
         return new WorkspaceEdit() { DocumentChanges = new[] { razorTextDocEdit } };
     }
 
-    private static string PopulateMethodSignature(VersionedDocumentContext documentContext, GenerateMethodCodeActionParams actionParams)
+    private async Task<string> PopulateMethodSignatureAsync(
+        VersionedDocumentContext documentContext,
+        GenerateMethodCodeActionParams actionParams,
+        string csharpSource,
+        OptionalVersionedTextDocumentIdentifier? codeBehindIdentifier,
+        CancellationToken cancellationToken)
     {
         var templateWithMethodSignature = s_generateMethodTemplate.Replace(s_methodName, actionParams.MethodName);
 
         var returnType = actionParams.IsAsync ? "System.Threading.Tasks.Task" : "void";
-        templateWithMethodSignature = templateWithMethodSignature.Replace(s_returnType, returnType);
 
         var eventTagHelper = documentContext.Project.TagHelpers
             .FirstOrDefault(th => th.Name == actionParams.EventName && th.IsEventHandlerTagHelper() && th.GetEventArgsType() is not null);
         var eventArgsType = eventTagHelper is null
             ? string.Empty // Couldn't find the params, generate no params instead.
-            : $"{eventTagHelper.GetEventArgsType()} e";
-
-        return templateWithMethodSignature.Replace(s_eventArgs, eventArgsType);
+            : eventTagHelper.GetEventArgsType();
+
+        var fullyQualifiedTypeNames = new[] { "System.NotImplementedException", returnType, eventArgsType };
```

Comment:
This shape of this API feels like an odd fit to me. I think it would make more sense for us to create the code for a method with fully qualified types, and send it to Roslyn saying "if I were to insert this code at position X in document Y, what should we actually insert?". That would be more future proof, and cover more bases than just type name qualifications. It could also mean we can do fancy things like generating with shortened type names, and inserting necessary using statements, or other simplifications that Roslyn might be able to do if we ever need to generate code that does something other than just throw.